tools/test-all

```diff
@@ -14,20 +14,51 @@ function run {
     fi
 }
 
-run ./tools/clean-repo
-run ./tools/test-tools
-run ./tools/lint-all --pep8
-run ./tools/test-migrations
-run ./tools/test-js-with-node
-run ./tools/run-mypy
-run ./tools/test-backend
-run ./tools/test-js-with-casper
-# Not running management test since it takes 40s and thus is too slow to be worth it.
-# run ./tools/test-management
-# Not running queue worker reload tests since it's low-churn code
-# run ./tools/test-queue-worker-reload
-# Not running documentation tests since it takes 20s and only tests documentation
-# run ./tools/test-documentation
-run ./tools/test-help-documentation.py
+function containsElement {
+    local e
+    for e in "${@:2}"; do [[ "$e" == "$1" ]] && return 0; done
+    return 1
+}
+
+TEST_COMMANDS=(
+    'run ./tools/check-provision' \
+    'run ./tools/clean-repo' \
+    'run ./tools/test-tools' \
+    'run ./tools/lint-all --pep8' \
+    'run ./tools/test-migrations' \
+    'run ./tools/test-js-with-node' \
+    'run ./tools/run-mypy' \
+    'run ./tools/test-backend' \
+    'run ./tools/test-js-with-casper' \
+    # Not running management test since it takes 40s and thus is too slow to be worth it.
+    # run ./tools/test-management
+    # Not running queue worker reload tests since it's low-churn code
+    # run ./tools/test-queue-worker-reload
+    # Not running documentation tests since it takes 20s and only tests documentation
+    # run ./tools/test-documentation
+    'run ./tools/test-help-documentation.py'
+)
+
+HAS_FORCE_ARGUMENT_TEST_COMMANDS=(
+    'run ./tools/lint-all --pep8' \
+    'run ./tools/test-backend' \
+    'run ./tools/test-js-with-casper'
+)
+
+if [[ $1 = "--force" ]]; then
```

Comment:
well, I think we should keep the original order ( [zulip chat](https://chat.zulip.org/#narrow/stream/development.20help/subject/tools.2Ftest-all.20order) )